components/gitpod-db/src/typeorm/migration/1664781308555-ChangeOauthCodePrimaryKey.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+const TABLE_NAME = "d_b_oauth_auth_code_entry";
+const COLUMN_NAME = "uid";
+
+export class ChangeOauthCodePrimaryKey1664781308555 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(
+            `ALTER TABLE ${TABLE_NAME} ADD COLUMN ${COLUMN_NAME} char(36) NOT NULL DEFAULT '', ALGORITHM=INPLACE, LOCK=NONE`,
+        );
+        await queryRunner.query(`UPDATE ${TABLE_NAME} SET ${COLUMN_NAME}=(SELECT uuid());`);
```

Comment:
Do we ever access the table by the generated PK? I'd have expected we'd set the `uid` column to the value in the PK

Reply:
In the [repository](https://github.com/gitpod-io/gitpod/blob/433b5680d19fd4994ebaadfb5e260dbe88c319a6/components/gitpod-db/src/typeorm/auth-code-repository-db.ts#L24-L73) we only access by [code](https://github.com/gitpod-io/gitpod/blob/433b5680d19fd4994ebaadfb5e260dbe88c319a6/components/gitpod-db/src/typeorm/auth-code-repository-db.ts#L38), never by `id`.

Reply:
Thanks!